bin/si-gen-cloud-control/src/pipeline-steps/generateActionFuncs.ts

```diff
@@ -0,0 +1,39 @@
+import { PkgSpec } from "../bindings/PkgSpec.ts";
+import _ from "lodash";
+import {
+  createActionFuncSpec,
+  createDefaultActionFuncs,
+} from "../spec/funcs.ts";
+import { ActionFuncSpecKind } from "../bindings/ActionFuncSpecKind.ts";
+
+export function generateDefaultActionFuncs(specs: PkgSpec[]): PkgSpec[] {
+  const newSpecs = [] as PkgSpec[];
+  const defaultActionFuncs = createDefaultActionFuncs();
+
+  for (const spec of specs) {
+    const schemaVariant = spec.schemas[0]?.variants[0];
+    const funcs = spec.funcs;
+    const actionFuncs = schemaVariant.actionFuncs;
+
+    if (!schemaVariant || !actionFuncs || !funcs) {
```

Comment:
Are actions funcs and funcs optional here? I though they were non optional arrays